tests/test_task_manager_memory_leak_guard.py

```
@@ -0,0 +1,477 @@
+            sock.connect(self.path)
+            sock.sendall((line + "\n").encode("utf-8"))
+            data = bytearray()
+            deadline = time.monotonic() + self.response_timeout
+            while b"\n" not in data:
+                remaining = deadline - time.monotonic()
+                if remaining <= 0:
+                    raise TimeoutError(f"timed out waiting for response to {line!r}")
+                sock.settimeout(remaining)
+                chunk = sock.recv(8192)
+                if not chunk:
+                    raise RuntimeError(f"socket closed while waiting for response to {line!r}")
```

Ruff (0.15.13)

[warning] Avoid specifying long messages outside the exception class

(TRY003)